iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitorTest.java

```diff
@@ -247,8 +247,11 @@ static Stream<Arguments> shouldReadProfileName() {
     return Stream.of(
       arguments("spring.profiles.active=profile1", "profile1"),
       arguments("spring.config.activate.on-profile=dev & qa", "dev & qa"),
-      arguments("spring.profiles.active=profile1\nspring.config.activate.on-profile=profile2", "profile1 profile2"),
-      arguments("#comment", ""));
+      arguments("spring.profiles.active=profile1\nspring.config.activate.on-profile=profile2", "profile2"),
+      arguments("spring.profiles.active=profile1\nspring.profiles.active=profile2", "profile2"),
+      arguments("spring.profiles.default=profile1\nspring.profiles.active=profile2", "profile2"),
+      arguments("spring.profiles.default=profile1", "profile1"),
+      arguments("#comment", "default"));
```

Comment:
Please add the following cases:
`"spring.config.activate.on-profile=profile1\nspring.profiles.active=profile2\n"`
`"spring.profiles.active=profile1\nspring.config.activate.on-profile=profile2\nspring.profiles.default=profile3"`

and maybe more.

Consider adding a space before/after `\n` for better readability.

Reply:
My spring crashes when I try to use `spring.profiles`.

I added the test cases.
I also added a whitespace after the profile name.
This led me to trim the profile name after reading it (confirmed in my spring application).

Adding a space before a line resulted in a parsingError, I created a ticket for it (https://sonarsource.atlassian.net/browse/SONARIAC-1457)